include/swift/Syntax/Serialization/SyntaxSerialization.h

```diff
@@ -302,13 +307,21 @@ struct ObjectTraits<syntax::RawSyntax> {
 
   static unsigned numFields(const syntax::RawSyntax &Syntax,
                             UserInfoMap &UserInfo) {
-    switch (nodeKind(Syntax, UserInfo)) {
-    case Token:
-      return 6;
-    case Layout:
-      return 5;
-    case Omitted:
-      return 2;
+    // FIXME: We know this is never set in production builds. Should we
+    // disable this code altogether in that case
+    // (e.g. if assertions are not enabled?)
```

Comment:
It would be nice if we could implement independent unit tests to test serialization/deserialization.